pkg/codeowners/owners.go

```diff
@@ -35,6 +40,34 @@ func NewFromPath(path string) ([]Entry, error) {
 	return ParseCodeowners(r), nil
 }
 
+func checkMultipleCodeownersFiles(dir string) error {
```

Comment:
A lot of the code in this method is a duplicate of the code in `openCodeownersFile`. I think it might be better to piggy back on that the `openCodeownersFile` to add functionality to detect multiple code owner files. Or perhaps do some refactoring of some sort to reduce code duplication.

Reply:
I'm 🤔 about this some more...

Reply:
What do you think about something like this:
```
func openCodeownersFile(dir string) (io.Reader, error) {
	codeownerFileCount := 0
	var fileReader *io.Reader = nil

	for _, p := range []string{".", "docs", ".github"} {
		pth := path.Join(dir, p)
		exists, err := afero.DirExists(fs, pth)
		if err != nil {
			return nil, err
		}
		if !exists {
			continue
		}
		f := path.Join(pth, "CODEOWNERS")
		_, err = fs.Stat(f)
		switch {
		case err == nil:
			codeownerFileCount++
		case os.IsNotExist(err):
			continue
		default:
			return nil, err
		}
		if fileReader != nil {
			return fmt.Errorf("Multiple CODEOWNERS files found in root, docs/, or .github/ directory of the repository %s", dir)
		}
		fileReader = fs.Open(f)
	}
	if fileReader != nil {
		return fileReader, nil
	}
	return nil, fmt.Errorf("No CODEOWNERS found in the root, docs/, or .github/ directory of the repository %s", dir)
}
```

Reply:
Cool, thanks! I started with this but had some trouble with values assignment and pointers .. all good now though